src/Extensions/Activities/Internal/ActivityObserver.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.ObjectPool;
+using Microsoft.Omex.Extensions.Abstractions.Activities.Processing;
+
+namespace Microsoft.Omex.Extensions.Activities
+{
+	internal class ActivityObserver : IActivityStartObserver, IActivityStopObserver
+	{
+		private readonly IActivitiesEventSender m_eventSender;
+		private readonly ILogger<ActivityObserver> m_logger;
+		private readonly ObjectPool<StringBuilder> m_stingBuilderPool;
+
+		public ActivityObserver(IActivitiesEventSender eventSender, ILogger<ActivityObserver> logger, ObjectPoolProvider objectPoolProvider)
+		{
+			m_eventSender = eventSender;
+			m_logger = logger;
+			m_stingBuilderPool = objectPoolProvider.CreateStringBuilderPool();
+		}
+
+		public void OnStart(Activity activity, object? payload = null) =>
+			m_logger.LogInformation(0x696e3739 /* tag_in79 */, $"Starting Activity {activity.OperationName} (Id:{activity.Id}; ParentId:{activity.ParentId})");
+
+		public void OnStop(Activity activity, object? payload = null)
+		{
+			m_eventSender.LogActivityStop(activity);
```

Comment:
What does this do and why is it logged before the Ending Activity message?
I'm assuming it is to do with ending the activity, whereas the other logging is just representing it in ULS logs.
If so, it would be a bit of an odd debugging experience to have the log message come after the ending of the activity in time.

Reply:
This probably require a better naming it actually sends an ETW event (or IFx in case of Lennon) that creates metrics.

Reply:
Renamed